components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/LoginDialogFragment.kt

```diff
@@ -71,7 +71,17 @@ internal class LoginDialogFragment : PromptDialogFragment() {
     private var username by SafeArgString(KEY_LOGIN_USERNAME)
     private var password by SafeArgString(KEY_LOGIN_PASSWORD)
 
-    private var stateUpdate: Job? = null
+    private var validateStateUpdate: Job? = null
+
+    /**
+     * These are essential to the function of this fragment.
+     */
+    private val loginValidationDelegate by lazy {
+        requireNotNull(requireNotNull(feature).loginValidationDelegate)
+    }
+
+    // List of potential dupes ignoring username
+    private var potentialDupesList: CopyOnWriteArrayList<Login>? = null
```

Comment:
whenever you use concurrency primitives, it's a really good idea to describe _why_, and what concerns you're trying to address. Otherwise, the only place this is currently documented is in the github discussion.